core/src/main/scala/cats/Parallel.scala

```diff
@@ -0,0 +1,164 @@
+
+  /**
+    * Natural Transformation from the parallel Applicative F[_] to the sequential Monad M[_].
+    */
+  def sequential: F ~> M
+
+  /**
+    * Natural Transformation from the sequential Monad M[_] to the parallel Applicative F[_].
+    */
+  def parallel: M ~> F
+
+  /**
+    * Provides an `ApplicativeError[F, E]` instance for any F, that has a `Parallel[M, F]`
+    * and a `MonadError[M, E]` instance.
+    * I.e. if you have a type M[_], that supports parallel composition through type F[_],
+    * then you can get `ApplicativeError[F, E]` from `MonadError[M, E]`.
+    */
+  def applicativeError[E](implicit E: MonadError[M, E]): ApplicativeError[F, E] =
+    Parallel.applicativeError[M, F, E](this, E)
+}
+
+object Parallel extends ParallelArityFunctions {
+
+  def apply[M[_], F[_]](implicit P: Parallel[M, F]): Parallel[M, F] = P
+
+  /**
+    * Like `Traverse[A].sequence`, but uses the applicative instance
+    * corresponding to the Parallel instance instead.
+    */
+  def parSequence[T[_]: Traverse, M[_], F[_], A]
+  (tma: T[M[A]])(implicit P: Parallel[M, F]): M[T[A]] = {
+    implicit val F = P.applicative
+    implicit val M = P.monad
+    val fta: F[T[A]] = Traverse[T].traverse(tma)(P.parallel.apply)
+    P.sequential(fta)
+  }
+
+  /**
+    * Like `Traverse[A].traverse`, but uses the applicative instance
+    * corresponding to the Parallel instance instead.
+    */
+  def parTraverse[T[_]: Traverse, M[_], F[_], A, B]
+  (ta: T[A])(f: A => M[B])(implicit P: Parallel[M, F]): M[T[B]] = {
+    implicit val F = P.applicative
+    implicit val M = P.monad
+    val gtb: F[T[B]] = Traverse[T].traverse(ta)(f andThen P.parallel.apply)
+    P.sequential(gtb)
+  }
+
+  /**
+    * Like `Foldable[A].sequence_`, but uses the applicative instance
+    * corresponding to the Parallel instance instead.
+    */
+  def parSequence_[T[_]: Foldable, M[_], F[_], A]
+  (tma: T[M[A]])(implicit P: Parallel[M, F]): M[Unit] = {
+    implicit val F = P.applicative
+    implicit val M = P.monad
+    val fu: F[Unit] = Foldable[T].traverse_(tma)(P.parallel.apply)
+    P.sequential(fu)
+  }
+
+  /**
+    * Like `Foldable[A].traverse_`, but uses the applicative instance
+    * corresponding to the Parallel instance instead.
+    */
+  def parTraverse_[T[_]: Foldable, M[_], F[_], A, B]
+  (ta: T[A])(f: A => M[B])(implicit P: Parallel[M, F]): M[Unit] = {
+    implicit val F = P.applicative
+    implicit val M = P.monad
+    val gtb: F[Unit] = Foldable[T].traverse_(ta)(f andThen P.parallel.apply)
+    P.sequential(gtb)
+  }
+
+  /**
+    * Like `Applicative[F].ap`, but uses the applicative instance
+    * corresponding to the Parallel instance instead.
+    */
+  def parAp[M[_], F[_], A, B](mf: M[A => B])
+                                    (ma: M[A])
+                                    (implicit P: Parallel[M, F]): M[B] = {
+    implicit val F = P.applicative
+    implicit val M = P.monad
+    val fb = Applicative[F].ap(P.parallel(mf))(P.parallel(ma))
```

Comment:
what about `P.applicative.ap()` rather than setting up the implicit to use it once.

Also, I don't see you using `val M` do we need it?